.circleci/config.yml

```diff
@@ -162,7 +181,7 @@ jobs:
             
 
 workflows:
```

Comment:
There's a lot of repetition between these workflows. I wonder whether that can be optimised easily

Reply:
we could possibly pull the jobs such as `linters`, `install_dependencies`, `run_specs` into a job of their own and call them all at once, however I think this would not really be an optimisation rather just DRYing the code. Which I dont think adds value here as when we deploy to main we want to be sure that all the individual sections run and pass before we deploy. It also allows for monitoring the progress on CircleCI as here: https://app.circleci.com/pipelines/github/ministryofjustice/laa-estimate-financial-eligibility-for-legal-aid/418/workflows/f7f12e13-d136-4ded-a877-128e098483e4
I think we could (if you wanted to) remove some of these checks from a deployment to main, after all the PR would have to pass all these before it could be merged. it adds duplication but also a level of assurance. Similarly we will repeat these steps for a deployment to staging and again for production, so we may want a wider discussion on what exactly our pipeline should look like.

Reply:
To clarify, I like the lots of small jobs. What I'm unsure about is having two workflows that mostly call all the same jobs, and whether we could have one workflow for both but with conditional steps (eg by branch filters etc) to handle where they differ.

It was the DRYing that I thought would be valuable. Just feels very vulnerable to going out of sync with each other, if something gets changed in one place but not the other.